src/marqo/tensor_search/filtering.py

```diff
@@ -0,0 +1,142 @@
+
+def build_tensor_search_filter(
+        filter_string: str, simple_properties: dict,
+        searchable_attribs: Sequence):
+    """Builds a Lucene-DSL filter string for OpenSearch, that combines the user's filter string
+    with searchable_attributes
+
+    """
+    if searchable_attribs is not None:
+        copied_searchable_attribs = copy.deepcopy(searchable_attribs)
+        searchable_attribs_filter = build_searchable_attributes_filter(
+            searchable_attribs=copied_searchable_attribs)
+    else:
+        searchable_attribs_filter = ""
+
+    filter_string_with_chunks_prefixes = add_chunks_prefix_to_filter_string_fields(
+        filter_string=filter_string, simple_properties=simple_properties)
+
+    if filter_string_with_chunks_prefixes and searchable_attribs_filter:
+        return f"({searchable_attribs_filter}) AND ({filter_string_with_chunks_prefixes})"
+    else:
+        return f"{searchable_attribs_filter}{filter_string_with_chunks_prefixes}"
+
+
+def build_searchable_attributes_filter(searchable_attribs: Sequence) -> str:
+    """Recursively constructs the filter used to narrow the search down to specific searchable attributes"""
+    if searchable_attribs is None or len(searchable_attribs) == 0:
+        return ""
+
+    vector_prop_count = len(searchable_attribs)
+
+    # brackets surround field name, in case it contains a space:
+    sanitised_attr_name = f"({sanitise_lucene_special_chars(searchable_attribs.pop())})"
+    
+    # base case
+    if vector_prop_count == 1:
+        return f"{enums.TensorField.chunks}.{enums.TensorField.field_name}:{sanitised_attr_name}"
+    else:
+        return (
+            f"{enums.TensorField.chunks}.{enums.TensorField.field_name}:{sanitised_attr_name}"
+            f" OR {build_searchable_attributes_filter(searchable_attribs=searchable_attribs)}")
+
+
+def sanitise_lucene_special_chars(to_be_sanitised: str) -> str:
+    """Santitises Lucene's special chars in a string.
+
+    We shouldn't apply this to the user's filter string, as they can choose to escape
+    Lucene's special chars themselves.
+
+    This should be used to sanitise a filter string constructed for users behind the
+    scenes (such as for searchable attributes).
+
+    See here for more info:
+    https://lucene.apache.org/core/6_0_0/queryparser/org/apache/lucene/queryparser/classic/package-summary.html#Escaping_Special_Characters
+
+    """
+    
+    # always escape backslashes before the other special chars
+    to_be_sanitised = to_be_sanitised.replace("\\", "\\\\")
+
+    # this prevents us from double-escaping backslashes.
+    non_backslash_chars = constants.LUCENE_SPECIAL_CHARS.union(constants.NON_OFFICIAL_LUCENE_SPECIAL_CHARS) - {'\\'}
+
+    for char in non_backslash_chars:
+        to_be_sanitised = to_be_sanitised.replace(char, f'\\{char}')
+    return to_be_sanitised
+
+
+def add_chunks_prefix_to_filter_string_fields(filter_string: Optional[str], simple_properties: typing.Iterable) -> str:
```

Comment:
if `filter_string` MUST NOT be `None`, it should be reflected in the signature:

Omit `Optional[]` and just leave `filter_string: str`

Reply:
filter string can be None, actually this is what is passed when the user does not input a filter string. simple_properties is the one that cannot be None.